### PR TITLE
Rewrite SafeTypes in Swift for compatibility

### PR DESCRIPTION
**Pull request by basine**
SafeTypes had to be rewritten because type checking didn't work anymore between String & NSString

### CODE REVIEW

File: Source/Public/Collections+ZMSafeTypes.swift [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+private let zmLog = ZMSLog(tag: "SafeTypes")
+
+
+func lastCallstackFrames() -> String {
+    let symbols = Thread.callStackSymbols
+    return symbols[0..<(min(7, symbols.count))].joined(separator: "\n")
+}
+
+
+func ObjectWhichIsKindOfClass<T>(dictionary: NSDictionary, key: String, required: Bool, transform: ((String) -> T?)?) -> T? {
```
Comment (marcoconti83) [inline Source/Public/Collections+ZMSafeTypes.swift]: 🔍🐜 Per convention, all functions name start with lowercase

File: Source/Public/Collections+ZMTSafeTypes.h [hunk 2]
```diff
@@ -21,30 +21,6 @@
 
 @class ZMEventID;
```
Comment (marcoconti83) [inline Source/Public/Collections+ZMTSafeTypes.h]: We don't need `ZMEventID` anymore here

**Comment by mikeger** [comment]
@basine I guess the tests for the methods are already there?

File: Source/Public/Collections+ZMTSafeTypes.h (hunk 2 above)
Comment (basine) [inline Source/Public/Collections+ZMTSafeTypes.h]: We still use it in Tests. Until we removed all references we can't remove it here.

**Comment by basine** [comment]
Tests were written in ObjC and are still there, yes

**Pull request merged**